src/backend/joanie/lms_handler/api.py

```diff
@@ -0,0 +1,101 @@
+"""
+API endpoints
+"""
+import hashlib
+import hmac
+
+from django.conf import settings
+
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+from joanie.core import models, utils
+from joanie.lms_handler import LMSHandler
+
+
+# pylint: disable=too-many-return-statements,unused-argument, too-many-locals,too-many-branches
+@api_view(["POST"])
+def course_runs_sync(request):
+    """View for the web hook to create or update course runs based on their resource link.
+
+    - A new course run is created or the existing course run is updated
+
+    Parameters
+    ----------
+    request : Type[django.http.request.HttpRequest]
+        The request on the API endpoint, it should contain a payload with course run fields.
+
+    Returns
+    -------
+    Type[rest_framework.response.Response]
+        HttpResponse acknowledging the success or failure of the synchronization operation.
+    """
+    msg = request.body.decode("utf-8")
+
+    # Check if the provided signature is valid against any secret in our list
+    #
+    # We need to do this to support 2 or more versions of our infrastructure at the same time.
+    # It then enables us to do updates and change the secret without incurring downtime.
+    authorization_header = request.headers.get("Authorization")
+    if not authorization_header:
+        return Response("Missing authentication.", status=401)
+
+    signature_is_valid = any(
+        authorization_header
+        == "SIG-HMAC-SHA256 {:s}".format(
+            hmac.new(
+                secret.encode("utf-8"),
+                msg=msg.encode("utf-8"),
+                digestmod=hashlib.sha256,
+            ).hexdigest()
+        )
+        for secret in getattr(settings, "JOANIE_COURSE_RUN_SYNC_SECRETS", [])
+    )
+
+    if not signature_is_valid:
+        return Response("Invalid authentication.", status=403)
```

Comment:
If I understand well `401` and `403` HTTP codes, they should be inverted.

`401` should be returned in case it lacks valid authentication credentials
`403` should be returned in case server refused to authorize the access to the resource. So in our case, when request is anonymous.

Reply:
Hmm I think you are right :thinking: